src/main/java/com/uid2/admin/vertx/service/SiteService.java

```diff
@@ -388,13 +377,33 @@ private void handleSiteUpdate(RoutingContext rc) {
                     ResponseUtil.error(rc, 400, "Invalid parameter for visible: " + visibleParam);
                 }
             }
+            if (name != null) {
+                if (validateSiteName(rc, name)) return;
+                existingSite.setName(name);
+            }
 
             uploadSiteToStoreWriterAndWriteExistingSiteToResponse(existingSite, rc);
         } catch (Exception e) {
             rc.fail(500, e);
         }
     }
 
+    private boolean validateSiteName(RoutingContext rc, String name) {
+        if (name == null || name.isEmpty()) {
+            ResponseUtil.error(rc, 400, "must specify a valid site name");
+            return true;
+        }
+
+        Optional<Site> existingSite = this.siteProvider.getAllSites()
+                .stream().filter(c -> c.getName().equals(name))
+                .findFirst();
+        if (existingSite.isPresent()) {
+            ResponseUtil.error(rc, 400, "site existed");
```

Comment:
nit: could we possibly align on some language and do something like we do in [ServiceService.java](https://github.com/IABTechLab/uid2-admin/blob/main/src/main/java/com/uid2/admin/vertx/service/ServiceService.java#L135)

So:
```
ResponseUtil.error(rc, 400, "site with name" + name + " already exists");
```